### ISSUE TITLE
create an easy way to auto-upgrade fabric8

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8#6186]
Right now I run this command:

```
#!/bin/sh
oc delete rc -l provider=fabric8
oc delete rs --all
oc delete route --all
oc delete deployment -l provider=fabric8
oc delete service -l provider=fabric8
oc delete ingress --all
oc delete secret -l provider=fabric8
oc delete configmap --all
oc delete build --all
oc delete buildconfig --all
oc delete sa -l provider=fabric8
```

then I do a redeploy

```
gofabric8 deploy -y
```

But that kinda sucks for end users! :) 

We should have some kind of update UI that shows if there are new versions of apps that you have deployed and allow either auto-upgrade or a manual upgrade of things

**Comment by minikiller** [comment]
I know under kubernetes there is rollingupdate, i suggested we should upgrade fabric8 pods in this way.

**Comment by jstrachan** [comment]
yeah; we're moving to the new fabric8-maven-plugin throughout fabric8 which means we'll release Deployments on kubernetes; or DeploymentConfig's on OpenShift. If we ever update those, then kubernetes/openshift automatically does rolling upgrade for us.

The main thing we need to get right first is move from OpenShift templates to using regular Deployment / DeploymentConfig resources for everything; then refer to ConfigMap values where we would have used OpenShift template parameters. Once thats done we should have all we need to update a Deployment / DeploymentConfig whenever we need to.

I was hoping helm would do this for us TBH https://github.com/kubernetes/helm - still worth investigating...

**Comment by jonasheinisch** [comment]
Is this still a thing? I guess as a first approach it would also be quite easy to use the above script and have it within gofabric8 as an "uninstall" command or sth...

**Comment by rawlingsj** [comment]
as it happens I'm actively working on these command until we have better helm update features

https://github.com/fabric8io/gofabric8/issues/178
and 
https://github.com/fabric8io/gofabric8/issues/177

Hopefully I'll have something by the end of the day if I get a clean run at it.

**Comment by jstrachan** [comment]
if you have persistence enabled you can do:

```
gofabric8 cleanup
gofabric8 start
```

the longer term [helm approach](http://fabric8.io/helm/) is going go be more like:
```
# install on vanilla k8s using helm:
helm init
helm repo add fabric8 https://fabric8.io/helm
helm install fabric8/fabric8-platform
...
# now upgrade
helm update fabric8
helm upgrade fabric8/fabric8-platform
```

**Comment by jstrachan** [comment]
this is now fixed in 2.4.24 with the gofabric8 CLI: http://fabric8.io/guide/getStarted/upgrading.html
